src/pages/home/ReportScreen.js

```diff
@@ -116,6 +121,21 @@ class ReportScreen extends React.Component {
         clearTimeout(this.loadingTimerId);
     }
 
+    /**
+     * When the report data is loaded, hide the ghost screen
+     *
+     * @static
+     * @param {Object} props
+     * @param {Object} state
+     * @return {Object|null}
+     */
+    static getDerivedStateFromProps(props, state) {
+        if (state.isGhostScreenVisible && props.isLoadingReportData === false) {
+            return {isGhostScreenVisible: false};
+        }
+        return null;
+    }
+
```

Comment:
We are migrating away from `getDerivedStateFromProps`. This can be done via `ComponentDidUpdate`.

Reply:
Generally I've avoided `setState` inside `componentDidUpdate`. Can do a conditional update.

Reply:
What is a conditional update? 
Create a function if linter throws an error for setState.